docs/design.md

```diff
@@ -0,0 +1,309 @@
+# having an immutable record that can be used during evaluation and runtime.
+
+# New 
+emporous-client create aggregate /path/to/config --freeze
+
+# Existing
+emporous-client create deployment-record registry.example.com/test:1.0
+...
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.index.v1+json",
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "size": 7143,
+      "digest": "sha256:e692418e4cbaf90ca69d05a66403747baa33ee08806650b51fab815ad7fc331f",
+    },
+     {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "size": 7023,
+      "digest": "sha256:ec4b8955958665577945c89419d1af06b5f7636b4ac3da7f12184802ad867736",
+      "attributes": {
+        "transitive": true
+      }
+    },
+  ]
+}
+```
+
+## Generate Software Inventory
+```bash
+emporous-client create inventory registry.example.com/test:1.0 --format spdx
+```
+
+## Consuming an artifact or aggregate
+```bash
+emporous-client pull registry.example.com/test:1.0 --follow-links
+```
+
+## Through the App Proxy
+```bash
+# Start app proxy server with CLI
+emporous-client serve /var/run/emporous.sock
+# With systemd
+systemctl start emporous-manager
+# With dnf plugin
+dnf install httpd
+# Find with fuse driver
+emporous-fuse ls
+...
+httpd
+```
+
+
+# High-Level Plan
+
+## Initial Approach
+
+1. Define an initial attribute extension for the OCI distribution specification
+2. Fork an OCI compliant registry and add functionality that will support the distribution specification
+3. Clarify extension specification following lessons learned from registry fork
+4. Develop Emporous client and smart proxy to interact with the attribute endpoint
+
+## Components
+
+## Distribution Spec Changes
+The goals for distribution spec changes are to build an extension to the existing OCI distribution spec to allow for cross-repo/cross-namespace attribute querying.
+
+With query optimization and backward compatibility in mind, we have the following requirements:
+- This must retain the V2 structure with standalone blob storage
+- This will need an indexing service to optimize certain types of queries (e.g. attributes)
+
+### RBAC
+The multi-tenant design of container registries will not change with the addition of attribute queries. Repository-scoped permissions should still be used and when returning artifact results, the result set should be filtered by what is viewable to the user. In other words, owners will be able to choose to make artifacts public.
+### Query scoping
+On the client side, scoping queries to namespaces and repositories will be supported.
+## CLI/Client Libraries
+We have existing CLI and library code bases that manage single and linked artifacts. These will continue to be used to publish single artifacts and manage local storage. The CLI/libraries will be extended to interface with the proposed API changes.
+## Service Layer
```

Comment:
Even if the service is running on a user's machine it is still a running service and is still acting as a gateway between the client and registry. Is there a better way this could be conveyed in your opinion? I do not believe this functionality to be optional.